pyvo/dal/tap.py

```diff
@@ -327,7 +327,14 @@ def run_async(
             self.baseurl, query, language=language, maxrec=maxrec, uploads=uploads,
             session=self._session, **keywords)
         job = job.run().wait()
-        job.raise_if_error()
+
+        try:
+            job.raise_if_error()
+        except DALQueryError as error:
+            if delete:
+                job.delete()
+            raise error
```

Comment:
you don't necessarily need to capture `error`, could just use raise as is
```suggestion
        except DALQueryError:
            if delete:
                job.delete()
            raise
```

Reply:
Wow, good to know, thanks :slightly_smiling_face: